internal/api/v1/stage.go

```diff
@@ -27,13 +31,87 @@ import (
 	"github.com/epinio/epinio/internal/application"
 	"github.com/epinio/epinio/internal/auth"
 	"github.com/epinio/epinio/internal/cli/clients/gitea"
+	"github.com/epinio/epinio/internal/duration"
 )
 
 const (
 	RegistryURL      = "registry.epinio-registry/apps"
 	DefaultInstances = int32(1)
 )
 
+// WaitUntilStaged will wait until the specified Tekton PipelineRun
+// has completed staging, sucessfully, or not, and returns the staging
+// result
+func (hc ApplicationsController) WaitUntilStaged(w http.ResponseWriter, r *http.Request) APIErrors {
+
+	ctx := r.Context()
+	log := tracelog.Logger(ctx)
+
+	params := httprouter.ParamsFromContext(ctx)
+
+	org := params.ByName("org")
+	name := params.ByName("app")
+	stageId := params.ByName("id")
+
+	log.Info("waiting until staging completes", "stageId", stageId)
+
+	cluster, err := kubernetes.GetCluster()
+	if err != nil {
+		return singleInternalError(err, "failed to get access to a kube client")
+	}
+
+	cs, err := tekton.NewForConfig(cluster.RestConfig)
+	if err != nil {
+		return singleError(err, http.StatusInternalServerError)
+	}
+	client := cs.TektonV1beta1().PipelineRuns(deployments.TektonStagingNamespace)
+
+	err = wait.PollImmediate(time.Second, duration.ToAppBuilt(),
+		func() (bool, error) {
+			l, err := client.List(context.TODO(),
```

Comment:
I think you should pass the request context here instead

Reply:
Will investigate, hopefully fix.

Reply:
Done, commit f25180ea1e97fd15e121ff23ab2390123ed23f0d